### PR TITLE
add tags to help generate openAPI schema

### PR DESCRIPTION
**Pull request by surajssd**
This will help with issue https://github.com/kedgeproject/kedge/issues/45 in helping generation of JSON schema.

### CODE REVIEW

**Comment by surajssd** [comment]
Blocked on https://github.com/kedgeproject/kedge/pull/239

**Comment by surajssd** [comment]
Also changes are based on #239

**Comment by surajssd** [comment]
TODOS:

- [ ] Add docs to what each of the tags mean
- [ ] Guide to developers who are gonna make changes to the spec.go

**Comment by cdrage** [comment]
@surajssd https://github.com/kedgeproject/kedge/pull/239 has been merged, just a docs issue now, right?

**Comment by kadel** [comment]
> TODOS:
> 
>  Add docs to what each of the tags mean
>  Guide to developers who are gonna make changes to the spec.go



Are going to address those TODOs in this PR or in another?
It would be good to have at least the first one in this PR.

**Comment by surajssd** [comment]
@kadel so I am working on this, with update of k8s 1.7 swagger schema I had to make some changes and now the generation is failing so I am trying to debug it, but if you wanna see how the changes will look like PTAL at https://github.com/surajssd/kedgeSchema#creating-jsonschema

**Comment by surajssd** [comment]
@cdrage @kadel @containscafeine open for review!

File: docs/development.md [hunk 1]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
 
 This will run only 4 tests in parallel. By default, it is set to the value of
 `GOMAXPROCS`.
+
+### spec.go conventions
+
+- Add explantion on top of each struct and struct field in `spec.go` to explain what it does,
```
Comment (cdrage) [inline docs/development.md]: explanation*

File: docs/development.md [hunk 2]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+so that when OpenAPI spec is auto-generated it shows up there.
```
Comment (cdrage) [inline docs/development.md]: `it will show up there`

File: docs/development.md [hunk 3]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+
+- Structs that are referred in any other struct in the form of array should have a comment
```
Comment (cdrage) [inline docs/development.md]: `in the form of an array`

File: docs/development.md [hunk 4]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+of the format `// kedgeSpec: io.kedge.*`, where `*` is name of that struct. This becomes the
+identity or refernce of that struct in OpenAPI specification.
```
Comment (cdrage) [inline docs/development.md]: reference

File: docs/development.md [hunk 5]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+
+- If in any struct you are embedding other struct there is no need to add explanation
```
Comment (cdrage) [inline docs/development.md]: rephrase

maybe: `If you are embedding a struct, there is no need to add an explanatory comment. `

File: docs/development.md [hunk 6]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+comment.
+
+- For any struct that is embedded please add a k8s tag comment which looks like this:
```
Comment (cdrage) [inline docs/development.md]: remove `which looks like this`

File: docs/development.md [hunk 7]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+`// k8s: io.k8s.kubernetes.pkg.api.v1.ServicePort`.
+
+- For all the fields that are optional please include a comment that looks like this:
```
Comment (cdrage) [inline docs/development.md]: remove `that looks like this`

File: docs/development.md [hunk 8]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+`// +optional`.
+
+- Any struct that is defined in same file and is used in other struct, while embedding
```
Comment (cdrage) [inline docs/development.md]: `used in another struct`

File: docs/development.md [hunk 9]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+please add a ref tag comment like following:
+`// ref: io.kedge.ContainerSpec`.
+
+- To find out what is the key or k8s reference tag for a particular struct in Kubernetes
```
Comment (cdrage) [inline docs/development.md]: missing comma: `struct in Kubernetes,`

File: docs/development.md [hunk 10]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+please refer the swagger specification of Kubernetes for any particular release. For e.g
```
Comment (cdrage) [inline docs/development.md]: `please refer to the`

File: docs/development.md [hunk 11]
```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
...
+In Kubernetes release 1.7 the reference tag for deployment is
```
Comment (cdrage) [inline docs/development.md]: `In Kubernetes 1.7, the reference tag for deployment is`

File: docs/development.md (hunk 1 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 2 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 3 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 4 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 6 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 7 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 8 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 9 above)
Comment (surajssd) [reply docs/development.md]: done

File: docs/development.md (hunk 10 above)
Comment (surajssd) [reply docs/development.md]: done

**Comment by cdrage** [comment]
LGTM

**Pull request merged**